lib/kitchen/verifier/pester.rb

```diff
@@ -369,42 +442,118 @@ def prepare_helpers
         end
       end
 
-      # Copies all test suite files into the suites directory in the sandbox.
+      # Creates a PowerShell hashtable from a ruby map.
+      # The only types supported for now are hash, string and Boolean.
       #
       # @api private
-      def prepare_pester_tests
-        info("Preparing to copy files from #{suite_test_folder} to the SUT.")
+      def ps_hash(obj, depth = 0)
+        if [true, false].include? obj
+          %{$#{obj}} # Return $true or $false when value is a bool
+        elsif obj.is_a?(Hash)
+          obj.map do |k, v|
+            # Format "Key = Value" enabling recursion
+            %{#{pad(depth + 2)}#{ps_hash(k)} = #{ps_hash(v, depth + 2)}}
+          end
+            .join("\n") # append \n to the key/value definitions
+            .insert(0, "@{\n") # prepend @{\n
+            .insert(-1, "\n#{pad(depth)}}\n") # append \n}\n
+
+        elsif obj.is_a?(Array)
+          array_string = obj.map { |v| ps_hash(v, depth + 4) }.join(",")
+          "#{pad(depth)}@(\n#{array_string}\n)"
+        else
+          # When the object is not a string nor a hash or array, it will be quoted as a string.
+          %{"#{obj}"}
```

Comment:
This could be dangerous if `obj` contains double quotes, as it won't be escaped.

Reply:
actually the `%{}` should handle that.

However `"#{obj}"` isn't needed and should just be `obj.to_s`

Reply:
It can also probably use an explicit comment here about how we're _relatively_ safe asusming all scalar values are strings, since PS will generally convert anything it needs to back into whatever type is has to be (especially for passing parameters).

Just in case it bites someone later on, would be good to call it out so they know what's going on. Easy enough to work around should it arise, but a comment will save some unneeded confusion.